File: sample apps/python/app/shared/execute_around_query.py
import time
import uuid


def timer(log_info):
    def partial(query, get_description):
        start = time.time()
        result = query()
        stop = time.time()
        duration = stop - start
        log_info("DURATION {} took {} milliseconds".format(get_description(), duration))
        return result

    return partial


def around(log_info):
    def partial(query, get_description):
        log_info("\nBEGIN {}".format(get_description()))
        query_time = timer(log_info)
        result = query_time(query, get_description)
        log_info("END {}\n".format(get_description()))
        return result

    return partial


def compensate(log_error, log_info):
    def partial(query, get_description):
        try:
            query_around = around(log_info)
            return query_around(query, get_description)
        except ValueError as ex:
            correlation_id = str(uuid.uuid4())
            log_info("Correlation Id {}".format(correlation_id))
            log_info("FAILED {}\n".format(get_description()))
            log_error(ex.message)
            raise ValueError("An error has occurred. Please contact support with correlation Id {}".format(correlation_id))

    return partial

File: sample apps/JavaScript/src/common/execute_around_query.js
import createUUID from './uuid';

export function logTime(logInfo) {
    return (query, getDescription) => {
        const start = new Date();
        const result = query();
        const end = new Date();        
        const duration = Math.floor(end - start) / (1000);
        logInfo(`DURATION: ${getDescription()} took ${duration}`); 
        return result;
    }
}

export function logWrap(logInfo) {
    return (query, getDescription) => {
        const query_timer = logTime(logInfo);
        logInfo(`BEGIN: ${getDescription()}`);
        const result = query_timer(query, getDescription);
        logInfo(`END: ${getDescription()}`);
        logInfo('\n');
        return result;
    }
}

export function handleError(logError, logInfo) {
    return (action, getDescription) => {
        const action_wrap = logWrap(logInfo);
        try {
            action_wrap(action, getDescription);
        } catch(ex) {
            logError(ex);
            const correlationId = createUUID();
            logInfo(`Correlation Id: ${correlationId}`);
            logInfo(`FAILED: ${getDescription()}`);
            logInfo('\n');
            throw new Error(`An error has occurred. Please contact support with correlation Id ${correlationId}`);
        }
    }
}